services/processor/native/adapter/native_compiler.go

```diff
@@ -90,15 +91,18 @@ func (c *nativeCompiler) Compile(ctx context.Context, code string) (*sdkContext.
 	hashOfCode := getHashOfCode(code)
 
 	writeTime := time.Now()
-	sourceCodeFilePath, err := writeSourceCodeToDisk(hashOfCode, code, artifactsPath)
+	sourceCodeFilePaths, err := writeSourceCodeToDisk(hashOfCode, code, artifactsPath)
 	c.metrics.writeToDiskTime.RecordSince(writeTime)
-	defer os.Remove(sourceCodeFilePath)
+	for _, sourceCodeFilePath := range sourceCodeFilePaths {
+		defer os.Remove(sourceCodeFilePath)
+	}
+
 	if err != nil {
```

Comment:
are you sure this is best practice ? it looks ugly but i understand it allows some of them to fail. 
if os.remove fails no one gets any log ...

Reply:
Even if some pollution happens, it's a temporary storage that we don't really care about.

Reply:
I'd like to see something slightly safer and more robust:
- hash each file separately, so that `getHashOfCode` returns `[]string`
- pass an array of structs to `writeSourceCodeToDisk` (something like `{Name: hash[i], Code: code[i[}`)
- since hash is unique for given code, you don't need to name using `.i`

WDYT?

Reply:
There is no need for that. Compilation already happens in a separate directory, `file.1.go`, `file.2.go` and so on are good enough. File names do not affect compilation in any way.